api/src/room_store/redis_room_store.py

```diff
@@ -101,58 +104,70 @@ class RedisRoomStore(RoomStore):
     def __init__(
         self,
         redis: Redis,
-        append_to_room_sha: str,
-        lreplace_sha: str,
-        delete_room_sha: str,
+        lreplace: Script,
+        delete_room: Script,
     ):
         self._redis = redis
-        self._append_to_room_sha = append_to_room_sha
-        self._lreplace_sha = lreplace_sha
-        self._delete_room_sha = delete_room_sha
+        self._lreplace = lreplace
+        self._delete_room = delete_room
         self._listeners_by_room_id: Dict[str, ChangeListener] = dict()
 
     async def _listen_for_changes(self, room_id: str) -> None:
         listener = self._listeners_by_room_id[room_id]
         queues = listener.output_queues
-        channel = (await self._redis.subscribe(_channel_key(room_id)))[0]
-        listener.subscribe_started.set_result(None)
-        try:
-            while await channel.wait_message():
-                update = from_dict(
-                    Request, json.loads((await channel.get()).decode('utf-8'))
-                )
+
+        async with self._redis.pubsub() as channel:
+            await channel.subscribe(_channel_key(room_id))
+
+            listener.subscribe_started.set_result(None)
+            async for msg in channel.listen():
+                if msg['type'] == 'subscribe':
+                    continue
+                update = from_dict(Request, json.loads(msg['data']))
                 for q in queues:
                     await q.put(update)
-        finally:
-            if not self._redis.closed:
-                await self._redis.unsubscribe(_channel_key(room_id))
+
+    def _make_done_callback(self, room_id: str) -> Callable[[Task], None]:
```

Comment:
This fixes a bug I noticed while testing the upgrade but isn't necessarily required for the upgrade. If the _listen_for_changes task failed, that room would just silently not get any updates until the server restarted. I can pull this change into a separate pr if you like.